packages/history-utility/src/history-utility.ts

```diff
@@ -40,11 +40,29 @@ export type History<T> = {
 
 type SubscribeOps = Parameters<Parameters<typeof subscribe>[1]>[0];
 
-export type HistoryOptions = {
+export type HistoryChangeMetadata = {
+  /**
+   * the operations that caused the history change
+   */
+  ops: SubscribeOps;
```

Comment:
I'd consider `ops` is sort of an internal/advanced api and wouldn't expose it like this. One can directly subscribe if they really want it.

In that sense, people can subscribe to proxyObject already, and this `onChange` is just a syntax sugar. That said, I'm fine if this improves DX.

Reply:
K will remove ops

Reply:
updated